src/types.ts

```diff
@@ -1,12 +1,12 @@
-export type SystemBarStyle = "light-content" | "dark-content";
+export type SystemBarStyle = "auto" | "light" | "dark";
 
-export type StatusBarProps = {
-  animated?: boolean;
-  barStyle: SystemBarStyle;
+export type SystemBarsEntry = {
+  statusBarHidden: boolean | undefined;
+  statusBarStyle: SystemBarStyle | undefined;
+  navigationBarHidden: boolean | undefined;
 };
 
-export type NavigationBarProps = {
-  barStyle: SystemBarStyle;
+export type SystemBarsProps = {
+  style?: SystemBarStyle;
+  hidden?: boolean | { statusBar?: boolean; navigationBar?: boolean };
```

Comment:
I wonder if this can become a type union itself. having hidden be undefined, a boolean, and a object with status bar and navigation bar feels...off, but it might be a nitpick. Dunno. Feel free to ignore if you don't see a better way

Reply:
This allow:

```tsx
<SystemBars
  hidden={true} // shortcut for { statusBar: true, navigationBar: true }
  // or:
  hidden={false} // shortcut for { statusBar: false, navigationBar: false }
  // or:
  hidden={{ statusBar: true }} // only hide status bar, keep navigation bar as it was before (might be visible or hidden)
  // or no hidden prop at all
/>
```